Fix to create all folders during webdav uploads

jdrossl opened the pull request:
https://github.com/craftercms/craftercms/issues/2581

jdrossl committed:
Fix to create all folders during webdav uploads

avasquez614 commented on src/main/java/org/craftercms/studio/impl/v1/service/webdav/WebDavServiceImpl.java:
```diff
@@ -180,22 +180,32 @@ public String upload(@ValidateStringParam(name = "site_id") final String site,
+                        sardine.createDirectory(uploadUrl);
+                        logger.debug("Folder created");
+                    } catch (Exception e) {
+                        logger.debug("Folder already exists");
```
Wouldn't it be better to first check if the dir exists and then create it? Instead of trying to create it and then ignore the exception if it already exists. Also what if the exception is a connection issue?

jdrossl replied:
This was implemented in this way because webdav only allows checking if files exist but not folders, so the only way was to try to create it and if an exception is thrown it means it already exists, about connectivity issues the current library doesn't provide specific exceptions to distinguish each case.

avasquez614 replied:
Understood. Maybe we can log the exception too and put a debug message of `Folder already exists or it couldn't be created`?

jdrossl replied:
Actually I think there is a way to distinguish them, I didn't notice because its a runtime exception, I will improve this

avasquez614 replied:
Ok good

avasquez614 replied:
Also when  logging add the `uploadUrl` to the message. Eg. `"Folder " + uploadUrl + " created"`

jdrossl committed:
Improvements for webdav:
- Better url handling to avoid missing/duplicated slashes
- Simpler checks for folders

jdrossl committed:
Improvements for debugging
